src/server/qgsserverprojectparser.cpp

```diff
@@ -644,69 +644,6 @@ QString QgsServerProjectParser::layerName( const QDomElement& layerElem ) const
   return nameElem.text().replace( QLatin1String( "," ), QLatin1String( "%60" ) ); //commas are not allowed in layer names
 }
 
-QString QgsServerProjectParser::serviceUrl() const
-{
-  QString url;
-
-  if ( !mXMLDoc )
-  {
-    return url;
-  }
-
-  QDomElement propertiesElement = propertiesElem();
-  if ( !propertiesElement.isNull() )
-  {
-    QDomElement wmsUrlElem = propertiesElement.firstChildElement( QStringLiteral( "WMSUrl" ) );
-    if ( !wmsUrlElem.isNull() )
-    {
-      url = wmsUrlElem.text();
-    }
-  }
-  return url;
-}
-
-QString QgsServerProjectParser::wfsServiceUrl() const
-{
-  QString url;
-
-  if ( !mXMLDoc )
-  {
-    return url;
-  }
-
-  QDomElement propertiesElement = propertiesElem();
-  if ( !propertiesElement.isNull() )
-  {
-    QDomElement wfsUrlElem = propertiesElement.firstChildElement( QStringLiteral( "WFSUrl" ) );
-    if ( !wfsUrlElem.isNull() )
-    {
-      url = wfsUrlElem.text();
-    }
-  }
-  return url;
-}
-
-QString QgsServerProjectParser::wcsServiceUrl() const
-{
-  QString url;
-
-  if ( !mXMLDoc )
-  {
-    return url;
-  }
-
-  QDomElement propertiesElement = propertiesElem();
-  if ( !propertiesElement.isNull() )
-  {
-    QDomElement wcsUrlElem = propertiesElement.firstChildElement( QStringLiteral( "WCSUrl" ) );
-    if ( !wcsUrlElem.isNull() )
-    {
-      url = wcsUrlElem.text();
-    }
-  }
-  return url;
-}
-
 void QgsServerProjectParser::combineExtentAndCrsOfGroupChildren( QDomElement& groupElem, QDomDocument& doc, bool considerMapExtent ) const
```

Comment:
A general consideration here: I think it's ok to get the service URLs from the project, but they are not mandatory (and should remain optional), just make sure that the server can build and provide sensible default URLs for at least the core services.
@dmarteau maybe we should try to address this in the 3.0 refactoring: every service should be able to build its own URL (with a sensible default method in the base class).